_posts/2018-03-01-Rust-1.24.1.md

```diff
@@ -0,0 +1,263 @@
+
+If you have a previous version of Rust installed via rustup, getting Rust
+1.24.1 is as easy as:
+
+```bash
+$ rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`][install] from the
+appropriate page on our website, and check out the [detailed release notes for
+1.24.1][notes] on GitHub.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/stable/RELEASES.md#version-1241-2018-03-01
+
+## What's in 1.24.1 stable
+
+Several minor regressions were found in 1.24.0 which collectively merited a
+release.
+
+A quick summary of the changes:
+
+* Do not abort when unwinding through FFI (this reverts behavior added in 1.24.0)
+* Emit UTF-16 files for linker arguments on Windows
+* Make the error index generator work again
+* Cargo will warn on Windows 7 if an update is needed.
+
+If your code is continuing to build, then the only issue that may affect you is
+the unwinding issue. We plan on bringing this behavior back in 1.25 or 1.26,
+depending on how smoothly the new strategy goes.
+
+With that, let's dig into the details!
+
+### Do not abort when unwinding through FFI
+
+TL;DR: the new behavior in 1.24.0 broke the `rlua` crate, and is being
+reverted. If you have since changed your code to take advantage of the behavior
+in 1.24.0, you'll need to revert it for now.  While we still plan to introduce
+this behavior eventually, we will be rolling it out more slowly and with a new
+implementation strategy.
+
+Quoting [the 1.24 annoucement](https://blog.rust-lang.org/2018/02/15/Rust-1.24.html):
+
+> There’s one other change we’d like to talk about here: undefined behavior.
+> Rust generally strives to minimize undefined behavior, having none of it in
+> safe code, and as little as possible in unsafe code. One area where you could
+> invoke UB is when a panic! goes across an FFI boundary. In other words, this:
+
+```rust
+extern "C" fn panic_in_ffi() {
+    panic!("Test");
+}
+```
+
+> This cannot work, as the exact mechanism of how panics work would have to
+> be reconciled with how the "C" ABI works, in this example, or any other ABI
+> in other examples.
+>
+> In Rust 1.24, this code will now abort instead of producing undefined behavior.
+
+As mentioned above, this caused breakage. It started with [a bug filed against
+the `rlua` crate](https://github.com/chucklefish/rlua/issues/71). `rlua` is a
+package that provides high level bindings between Rust and the [Lua programming
+language](https://www.lua.org/).
+
+> Side note: `rlua` is maintained by [Chucklefish](https://chucklefish.org/),
+> a game development studio from London that's using Rust. Lua is a very
+> popular language to use for extending and scripting games. We care deeply about
+> production Rust users, and so handling this was a very high priority for the
+> Rust team.
+
+On Windows, and only on Windows, any attempt to handle errors from Lua would
```

Comment:
Probably not important, but this is MSVC-specific (since MinGW doesn't use SEH).

Reply:
Seems important to me, since people use MinGW and might want to know!